pkg/config/structure/unmarshal.go

```diff
@@ -369,6 +370,32 @@ func copyList(target reflect.Value, inputList []nodetreemodel.Node, currPath []s
 
 func copyAny(target reflect.Value, input nodetreemodel.Node, currPath []string, fs *featureSet) error {
 	if target.Kind() == reflect.Pointer {
+		if target.Type().Elem().Kind() != reflect.Invalid && isScalarKind(reflect.New(target.Type().Elem()).Elem()) {
```

Comment:
I don't think this block of code is necessary. It is performing extra allocations and adds lots of complexity that isn't needed. If it's removed, the tests that used to be failing (before this PR) will still succeed because the actual issue was the usage of structs being passed to `SetWithoutSource`.

It does fix a real problem, that of assigning nil to fields of a struct that aren't already nil, but I think we should hold off on that, and save it for another PR, because we want to ensure the behavior is the same as what viper did.